src/libraries/Microsoft.PowerFx.Json/FormulaValueJSON.cs

```diff
@@ -39,28 +39,38 @@ public static FormulaValue FromJson(string jsonString, FormulaValueJsonSerialize
                 return FromJson(propBag, settings, formulaType);
             }
             catch (JsonException je)
-            {                
+            {
                 return new ErrorValue(IRContext.NotInSource(formulaType), new ExpressionError()
                 {
                     Message = $"{je.GetType().Name} {je.Message}",
                     Span = new Syntax.Span(0, 0),
                     Kind = ErrorKind.Network
                 });
             }
+            catch (PowerFxJsonException pfxje)
+            {
+                return new ErrorValue(IRContext.NotInSource(formulaType), new ExpressionError()
+                {
+                    Message = $"{pfxje.GetType().Name} {pfxje.Message}",
+                    Span = new Syntax.Span(0, 0),
+                    Kind = ErrorKind.Network

```

Comment:
ideally we would love to have a different kind but if it's too hard I think we can live without it as well

Reply:
Will add ErrorKind.InvalidJson here